src/macros.rs

```diff
@@ -56,36 +55,76 @@ macro_rules! run_output {
 #[macro_export]
 macro_rules! run_result {
     ($($args:tt)*) => {{
-        let context = $crate::context::Context::production();
-        $crate::run_result_with_context!(context, $($args)*)
+        $crate::input::Input::run_result($crate::tuple_up!($($args)*))
     }}
 }
 
 #[doc(hidden)]
 #[macro_export]
-macro_rules! run_result_with_context {
-    ($context:expr, $($args:tt)*) => {{
-        let mut config = $crate::config::Config::default();
-        $crate::configure!(config: config, args: $($args)*);
-        $crate::child_output::ChildOutput::run_child_process_output($context, config)
-    }}
-}
-
-#[doc(hidden)]
-#[macro_export]
-macro_rules! configure {
-    (config: $config:ident, args: % $head:expr $(,)?) => {
-        $crate::input::Input::configure($crate::input::Split($head), &mut $config);
+macro_rules! tuple_up {
```

Comment:
The diff for `tuple_up!` looks pretty bad. Probably easier to look at this: https://github.com/soenkehahn/cradle/blob/macros-in-terms-of-tuples/src/macros.rs#L62..L130